test/parallel/test-tls-socket-close.js

```diff
@@ -45,24 +49,20 @@ function connectClient(server) {
     rejectUnauthorized: false
   });
 
-  clientTlsSocket.write('foo', 'utf8', common.mustCall(() => {
-    assert(netSocket);
-    netSocket.setTimeout(common.platformTimeout(10), common.mustCall(() => {
-      assert(serverTlsSocket);
+  clientTlsSocket.write('.');
 
-      netSocket.destroy();
-      assert.strictEqual(netSocket.destroyed, true);
+  clientTlsSocket.on('data', (chunk) => {
+    assert.strictEqual(chunk[0], 46);
 
-      setImmediate(() => {
-        // Close callbacks are executed after `setImmediate()` callbacks.
-        assert.strictEqual(netSocketCloseEmitted, false);
-        assert.strictEqual(serverTlsSocket.destroyed, false);
-        setImmediate(() => {
-          assert.strictEqual(netSocketCloseEmitted, true);
-        });
-      });
-    }));
-  }));
+    netSocket.destroy();
+    assert.strictEqual(netSocket.destroyed, true);
+
+    setImmediate(() => {
```

Comment:
No `common.mustCall` here?

Reply:
I don't really get what you mean, `common.mustCall` wouldn't be there to validate `setImmediate` behavior, but to ensure the assertion are actually run. Anyway, it doesn't really matter.

Reply:
We are testing that the callback of `setImmediate()` in other tests.

Reply:
It's not called if e.g. `process.exit` is called before it's picked up

Reply:
Adding `common.mustCall` would ensure we catch it if it was ever added

Reply:
I've added it only where it is needed given the current assumptions. It the logic changes there are other more places where it would be needed.